common/buildcraft/factory/tile/TileMiningWell.java

```diff
@@ -95,7 +95,7 @@ private boolean canBreak() {
     }
 
     private void nextPos() {
-        for (currentPos = pos.down(); currentPos.getY() >= 0; currentPos = currentPos.down()) {
+        for (currentPos = pos.down(); currentPos.getY() >= CubicChunksChecker.worldMin; currentPos = currentPos.down()) {
```

Comment:
Assuming the start position is not above the world (which seems reasonable), you could use `world.isOutsideBuildHeight()`.